### PR TITLE
[1.13] ci: Add IPsec leak detection for ci-ipsec-e2e

### PR DESCRIPTION
**Pull request by jschwinger233**
- [ ] https://github.com/cilium/cilium/pull/32930 @jschwinger233 

```upstream-prs
 32930
```

### CODE REVIEW

**Comment by jschwinger233** [comment]
/test-backport-1.13

**Comment by jschwinger233** [comment]
/test-1.17-4.19

**Comment by jschwinger233** [comment]
/test-1.17-4.19

**Comment by julianwiedmann** [comment]
/test-1.17-4.19

**Comment by julianwiedmann** [comment]
//test-1.17-4.19

**Comment by julianwiedmann** [comment]
/test-1.17-4.19

**Comment by julianwiedmann** [comment]
@jschwinger233 oh wow, just need to try often enough :sweat_smile:

File: .github/workflows/conformance-ipsec-e2e.yaml [hunk 1]
```diff
@@ -307,7 +307,7 @@ jobs:
         uses: ./.github/actions/bpftrace/start
         with:
           script: ./.github/actions/bpftrace/scripts/check-ipsec-leaks.bt
-          args: ${{ steps.bpftrace-params.outputs.params }} "true"
+          args: ${{ steps.bpftrace-params.outputs.params }} "false"
```
Comment (julianwiedmann) [inline .github/workflows/conformance-ipsec-e2e.yaml:310]: I think we should have some more detailed context flying around on why v1.13 needs this. I'll try do dig it up, and open a follow-up PR to add a comment here :+1:.

**Review by julianwiedmann — APPROVED** [review]

File: .github/workflows/conformance-ipsec-e2e.yaml (hunk 1 above)
Comment (jschwinger233) [reply .github/workflows/conformance-ipsec-e2e.yaml:310]: I forgot either, but I found the backporter's note:

```
[ backporter's note: In 1.13 ci-ipsec-e2e disables IPv6, we need to
suppress "no IPv6 connections" errors in check-ipsec-leaks.bt ]
```

Comment (julianwiedmann) [reply .github/workflows/conformance-ipsec-e2e.yaml:310]: I found this excellent summary by @giorio94. Adding it here so I find it for the follow-up PR :).

> Endpoints routes appear to be broken with IPv6 on Cilium <1.14 when
> there are netpols installed [1,2], and the Cilium CLI thus skips
> these test cases [3]. Let's update the IPSec unencrypted packets
> checks to not error out if we don't see IPv6 packets originated
> from the L7 proxy, as expected in this case.
> 
> 1: https://github.com/cilium/cilium/issues/23852
> 2: https://github.com/cilium/cilium/issues/23910
> 3: https://github.com/cilium/cilium-cli/blob/5df839f3a77d5c7ce5efa1cff6066438ee0c3206/connectivity/check/test.go#L897-L905

**Review by aanm — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]

**Review by nebril — APPROVED** [review]

File: .github/workflows/conformance-ipsec-e2e.yaml (hunk 1 above)
Comment (julianwiedmann) [reply .github/workflows/conformance-ipsec-e2e.yaml:310]: Continuing here: https://github.com/cilium/cilium/pull/33576

**Pull request merged**